src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.unit.netcoreapp.cs

```diff
@@ -0,0 +1,332 @@
+            Setup(watcher);
+
+            watcher.Filters.Remove("*.pdb");
+            Assert.DoesNotContain(watcher.Filters, t => t == "*.pdb");
+            EqualItems(new Collection<string> { "*.dll" }, watcher.Filters);
+
+            // No Exception is thrown while removing an item which is not present in the list.
+            watcher.Filters.Remove("*.pdb");
+        }
+
+        [Fact]
+        public void AddNullOrEmptyStringToFilters()
+        {
+            var watcher = new FileSystemWatcher();
+            Setup(watcher);
+
+            watcher.Filters.Add(string.Empty);
+            Assert.Equal(0, watcher.Filters.Count);
+
+            Setup(watcher);
+            Assert.Equal(2, watcher.Filters.Count);
+            watcher.Filters.Add(null);
+            Assert.Equal(0, watcher.Filters.Count);
+            EqualItems(new Collection<string> { }, watcher.Filters);
+        }
+
+        [Fact]
+        public void SetNullOrEmptyStringToFilters()
+        {
+            var watcher = new FileSystemWatcher();
+            Setup(watcher);
+
+            watcher.Filters[0] = string.Empty;
+            Assert.Equal(0, watcher.Filters.Count);
+
+            Setup(watcher);
+            Assert.Equal(2, watcher.Filters.Count);
+            watcher.Filters[0] = null;
+            Assert.Equal(0, watcher.Filters.Count);
+            EqualItems(new Collection<string> { }, watcher.Filters);
+        }
+
+        [Fact]
+        public void ClearFilters()
+        {
+            var watcher = new FileSystemWatcher();
+            Setup(watcher);
+
+            watcher.Filters.Clear();
+            Assert.Equal(0, watcher.Filters.Count);
+            EqualItems(new Collection<string> { }, watcher.Filters);
+        }
+
+        [Fact]
+        public void GetFilterAfterFiltersClear()
+        {
+            using (var testDirectory = new TempDirectory(GetTestFilePath()))
+            {
+                var watcher = new FileSystemWatcher(testDirectory.Path);
+                Setup(watcher);
+
+                watcher.Filters.Clear();
+                Assert.Equal("*", watcher.Filter);
+                EqualItems(new Collection<string> { }, watcher.Filters);
+            }
+        }
+
+        [Fact]
+        public void GetFiltersAfterFiltersClear()
+        {
+            using (var testDirectory = new TempDirectory(GetTestFilePath()))
+            {
+                var watcher = new FileSystemWatcher(testDirectory.Path);
+                Setup(watcher);
+
+                watcher.Filters.Clear();
+                Assert.Throws<ArgumentOutOfRangeException>(() => watcher.Filters[0]);
+                Assert.Equal(0, watcher.Filters.Count);
+                Assert.Empty(watcher.Filters);
+                Assert.NotNull(watcher.Filters);
+            }
+        }
+
+        [Fact]
+        public void InvalidOperationsOnFilters()
+        {
+            var watcher = new FileSystemWatcher();
+            Setup(watcher);
```

Comment:
This helper makes it hard to follow what the test is doing. It isn't doing anything but add two items, just do it here. Same goes for other usages. If you want to assert what that helper does currently, add a specific test.